### PR TITLE
Update pants homebrew cask on release.

### PR DESCRIPTION
**Pull request by kaos**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -139,3 +139,26 @@ jobs:
...
+      - name: Bump `scie-pants` version in Casks/pants.rb to ${{ needs.determine-tag.outputs.release-tag }}
+        run: |
+          ./bump_version.sh ${{ needs.determine-tag.outputs.release-version }}
+
+          git config user.name github-actions
+          git config user.email github-acctions@github.com
+
+          git add Casks/pants.rb
+          git commit -m "Bump scie-pants ${{ needs.determine-tag.outputs.release-tag }}"
+          git push
```
[the remark this answers is not in the document]
Comment (kaos) [reply .github/workflows/release.yml:164]: Regarding perms. I was looking at the example here: https://github.com/actions/checkout#push-a-commit-using-the-built-in-token

I've no idea how this really works, and are certainly not sure if the GITHUB_TOKEN would be usable to access other repositories like I do here.

I can see having a workflow over in `pantsbuild/homebrew-tap` that does this make sense and potentially easier to test/try out, will just have to figure out how to kick that off from here then.

[the remark this answers is not in the document]
Comment (jsirois) [reply .github/workflows/release.yml:164]: Ok, yeah. I guess maybe that example push makes sense for the current repo. I have less confidence in a foreign repo push. You can find out with an experiment on a personal repo & pair of repos pretty easy though.

[the remark this answers is not in the document]
Comment (kaos) [reply .github/workflows/release.yml]: Yea, I've added the bump workflow over in pantsbuild/homebrew-tap#11 

Tweaked the release workflow here to trigger that one, just need a `TAP_TOKEN` secret setup and it "should" work, I hope :)

[the remark this answers is not in the document]
Comment (jsirois) [reply .github/workflows/release.yml]: Great, that's a better structure I think. Tell me what you need re TAP_TOKEN. Do you need elevated privs to self serve? If not how do I obtain the value of TAP_TOKEN to set?

[the remark this answers is not in the document]
Comment (jsirois) [reply .github/workflows/release.yml]: I just added you to the admin role to take the guesswork out. You should be able to add the secret with your PAT on your own now.

**Review by jsirois — APPROVED** [review]

**Review by jsirois — APPROVED** [review]

**Pull request merged**